### PR TITLE
Deployments Http impls

### PR DESCRIPTION
**Pull request by andrewazores**
This set of commits replaces most of the DeploymentsServices functions with implementations that perform queries to the backend WIT endpoint, rather than returning mock data. This brings the UI up to speed with the current state of the API provided by the endpoint as of the time of writing.

Notable bugs:
- network sparkline is still mocked - no backend support
- dropdown menu items (View logs, Open Application, etc.) are still mocked - no backend support
- pods donut does not support all possible pod states, and can incorrectly display legend items for pod states that are not actually represented in the data. The next version of the backend API has a more suitable response format, so I've simply taken a "good enough" shortcut for the time being
- memory sparkline does not seem to respect the "total" amount on the graph. For my own personal example deployment, the sparkline is pegged at 100% even though the data shows 151.4/1024 MB.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2404-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2404-fabric8-ui.badger.fabric8.io

File: src/app/space/create/deployments/services/deployments.service.spec.ts [hunk 1]
```diff
@@ -147,126 +153,233 @@ describe('DeploymentsService', () => {
...
+              name: 'vertx-hello',
+              pipeline: [
+                {
+                  name: 'run'
+                }
+              ]
+            }
+          ]
+        }
+      };
```
Comment (jiekang) [inline src/app/space/create/deployments/services/deployments.service.spec.ts]: On second thought, could we get a few more test cases against empty and multiple spaces/applications/environments?

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2404-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2404-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2404-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2404-fabric8-ui.badger.fabric8.io

**Comment by jiekang** [comment]
I've tested against prod-preview with some 'empty' application deployments. Things looked sane and no more null issue breaking the entire application.
